docs: Add documentation for large version jump upgrades (#8678)
hashicorp/consul: website/pages/docs/upgrading/index.mdx

   by running `consul members` to make sure all members have the latest
   build and highest protocol version.

## Large Version Jumps

> Remaining many major versions behind can increase the risk incured during

Reviewer: ```suggestion
Operating a Consul datacenter that is multiple major versions behind the current major version can increase the risk incured during
```

> Operating a Consul datacenter that is multiple major versions behind the current major version can increase the risk incured during

Reviewer 2: typo, should be incurred

> upgrades. We encourage our customers to remain no more than two major versions behind

Reviewer 2: We encourage our users to..

> (i.e., if 1.8.x is the current release, do not use versions older than 1.6.x),

Reviewer: ```suggestion
(i.e., if 1.8.x is the current release, do not use versions older than 1.6.x).
```

> but in some environments that is not possible. If you find yourself in a situation

Reviewer: ```suggestion
If you find yourself in a situation
```

   by running `consul members` to make sure all members have the latest
   build and highest protocol version.

## Large Version Jumps

Consul is a service that frequently sits in the critical path for many
or even all other applications and services you might have. As such,
it can be easy to get into a "if it's not broken, don't fix it" mentality
to avoid the potential risk in upgrading. Unfortunately, this can lead to
being many major versions behind and ultimately just makes upgrading
riskier overall.

We encourage our customers to remain no more than two major versions behind
(i.e., if 1.8.x is the current release, don't use versions older than 1.6.x),
but in some environments that's not possible. If you find yourself in this

> situation and need to upgrade, please see our [Upgrade Instructions page](/docs/upgrading/instructions) for

Reviewer: ```suggestion
situation, and need to upgrade, please see our [Upgrade Instructions page](/docs/upgrading/instructions) for
```
